frontend/packages/ux-editor/src/containers/DesignView/PageGroupAccordion.test.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { screen, within } from '@testing-library/react';
+import type { PagesModel } from 'app-shared/types/api/dto/PagesModel';
+import { renderWithProviders } from '../../testing/mocks';
+import { PageGroupAccordion, type PageGroupAccordionProps } from './PageGroupAccordion';
+import { layoutSet1NameMock } from '../../testing/layoutSetsMock';
+import type { IFormLayouts } from '../../types/global';
+import { layout1NameMock, layoutMock } from '../../testing/layoutMock';
+import { createQueryClientMock } from 'app-shared/mocks/queryClientMock';
+import { app, org, pageGroupAccordionEllipsisMenu } from '@studio/testing/testids';
+import { QueryKey } from 'app-shared/types/QueryKey';
+import userEvent from '@testing-library/user-event';
+import { textMock } from '@studio/testing/mocks/i18nMock';
+import type { ServicesContextProps } from 'app-shared/contexts/ServicesContext';
+
+const pagesMock: PagesModel = {
+  pages: null,
+  groups: [
+    {
+      name: 'Group 1',
+      order: [{ id: 'Side 1' }],
+    },
+    {
+      name: 'Group 2',
+      order: [{ id: 'Side 2' }],
+    },
+  ],
+};
+
+const layoutSetName = layoutSet1NameMock;
+const layouts: IFormLayouts = {
+  [layout1NameMock]: layoutMock,
+};
+
+describe('PageGroupAccordion', () => {
+  it('should disable move-up for first group, and move-down for last group', async () => {
+    await render({});
+    expect(moveGroupUpButton(0)).toBeDisabled();
+    expect(moveGroupDownButton(0)).toBeEnabled();
+    expect(moveGroupUpButton(1)).toBeEnabled();
+    expect(moveGroupDownButton(1)).toBeDisabled();
+  });
+
+  it('should correctly call mutation on changePageGroupOrder when moving a group up', async () => {
+    const user = userEvent.setup();
+    const changePageGroups = jest.fn();
+    await render({ queries: { changePageGroups } });
+    await user.click(moveGroupUpButton(1));
+    expect(changePageGroups).toHaveBeenCalledTimes(1);
+    const expectedPagesMock = pagesMock;
+    expectedPagesMock.groups = pagesMock.groups.toReversed();
```

Comment:
```suggestion
const expectedPagesMock = {
  ...pagesMock,
  groups: pagesMock.groups.toReversed(),
};
```
Nitpicking, but the original solution creates a reference to `pagesMock` and not actually a copy. 
It will be no concequences in this case, so it is not necessary to change.

Reply:
Ah you are right, I chose to use `toReversed` to avoid changing the reference, so good catch! I will change this, to avoid causing issues with other tests.